src/workerd/jsg/resource.h

```diff
@@ -597,6 +597,142 @@ class DynamicResourceTypeMap {
   friend class ObjectWrapper;
 };
 
+// ======================================================================================
+// NamedIntercept implementation
+
+class JsValue;
+
+// A utility class that allows dynamic evaluation of properties on the JavaScript wrapper.
+// TODO(soon): Currently, this mechanism only supports read-only properties. The setter
+// interceptor is not implemented.
```

Comment:
To add clarification here: "_The setter interceptor is not implemented_" means for a dynamic name like `foo` (in this example), I can't set `obj.foo` in JS. 

However, I _can_ set `obj.baz` because it would return `kj::none` in `queryNamed()`.

In the case of JS RPC, we always return `kj::none` for `queryNamed()` and an empty array for `listNamed()`, so no properties are read-only. If I define a method `foo()` on my remote DO, then I can overwrite `stub.foo = {/*whatever*/}`.

Is that alright @irvinebroque?

Reply:
Expanded the comments here.